src/main/java/de/codefor/opacapi/OpacAPI.java

```diff
@@ -59,8 +60,12 @@ public static void main(final String[] args) throws JSONException, OpacApi.OpacE
         System.out.println("First match: " + searchRequestResult.getResults().get(0).toString());
 
         System.out.println("Fetching details for the first result...");
-        DetailedItem detailedItem = api.getResult(0);
-        System.out.println("Got details: " + detailedItem.toString());
+        if (api instanceof Open)
+            System.out.println("detailedItem item not supported:-(");
```

Comment:
Getting the detailled results is in fact supported for all APIs, but in some cases you need to use `getResultById` with the ID provided in the search result instead of `getResult` with the position in the search results. In our app, we use `getResultById` by default and `getResult` as a fallback.

Reply:
Thank you for the explanation!